src/modules/SdnDiag.Utilities/private/Export-ObjectToFile.ps1

```diff
@@ -32,35 +32,37 @@ function Export-ObjectToFile {
 
     begin {
         $arrayList = [System.Collections.ArrayList]::new()
-
-        # if object is null, then exit
-        if ($null -eq $Object) {
-            return
+        # build the file directory and name that will be used to export the object out
+        if($Prefix){
+            [System.String]$formattedFileName = "{0}\{1}_{2}.{3}" -f $FilePath.FullName, $Prefix, $Name, $FileType
         }
-    }
-    process {
-        foreach ($obj in $Object) {
-            [void]$arrayList.add($obj)
+        else {
+            [System.String]$formattedFileName = "{0}\{1}.{2}" -f $FilePath.FullName, $Name, $FileType
         }
-    }
-    end {
-        try {
-            # build the file directory and name that will be used to export the object out
-            if($Prefix){
-                [System.String]$formattedFileName = "{0}\{1}_{2}.{3}" -f $FilePath.FullName, $Prefix, $Name, $FileType
-            }
-            else {
-                [System.String]$formattedFileName = "{0}\{1}.{2}" -f $FilePath.FullName, $Name, $FileType
-            }
+        [System.IO.FileInfo]$fileName = $formattedFileName
 
-            [System.IO.FileInfo]$fileName = $formattedFileName
+        # create the parent directory structure if does not already exist
+        if(!(Test-Path -Path $fileName.Directory -PathType Container)){
 
-            # create the parent directory structure if does not already exist
-            if(!(Test-Path -Path $fileName.Directory -PathType Container)){
+            try {
                 "Creating directory {0}" -f $fileName.Directory | Trace-Output -Level:Verbose
                 $null = New-Item -Path $fileName.Directory -ItemType Directory
             }
+            catch {
+                throw New-Object System.Exception("Failed to create directory $($fileName.Directory)")
```

Comment:
>throw New-Object System.Exception("Failed to create directory $($fileName.Directory)") [](http://example.com/codeflow?start=16&length=86)

Why do this ? This would mask the original exception right ?

Reply:
I can update to include the original error message